src/isomorphic/classic/element/ReactElement.js

```diff
@@ -245,7 +245,8 @@ ReactElement.cloneElement = function(element, config, children) {
     // Remaining properties override existing props
     for (propName in config) {
       if (config.hasOwnProperty(propName) &&
-          !RESERVED_PROPS.hasOwnProperty(propName)) {
+          !RESERVED_PROPS.hasOwnProperty(propName) &&
+          config[propName] !== undefined) {
```

Comment:
This is wrong, because setting the value to undefined SHOULD change the element if the prop was set to a non-default value when the element was created.  The desired behavior would be to use the prop's default value.